grammar/TypeQL.g4

```diff
@@ -161,24 +168,40 @@ predicate             :   value
 predicate_equality    :   EQ | NEQ | GT | GTE | LT | LTE ;
 predicate_substring   :   CONTAINS | LIKE ;
 
-predicate_value       :   value | VAR_  ;
+predicate_value       :   value | VAR_CONCEPT_  | VAR_VALUE_ ;
+
+// EXPRESSION CONSTRUCTS =======================================================
 
-// SCHEMA CONSTRUCT =============================================================
+expression                  :   expression_operation    |   expression_structure        ;
+expression_operation        :   <assoc=right> expression_operation POW expression_operation         // exponentiation is right-associative
+                            |   expression_operation  (TIMES | DIV | MOD)  expression_operation
+                            |   expression_operation  (PLUS  | MINUS) expression_operation
+                            |   expression_structure
+                            ;
+expression_structure        :   '(' expression ')'
```

Comment:
I don't think expression_structure is the right name for this. @flyingsilverfin

Reply:
I don't think it's quite right either, but let's keep it for now. I just noticed that both `expression` and `expression_operation` fall through to `expression_structure` - is this ambiguous? Can you get rid of `expression_structure` inside `expression_operation`?

Reply:
Yes. That also gives us a redundant expression : expression_operation
So we can collapse `expression`, `expression_operation` and `expression_structure` the whole thing into a single rule. Shall we?

Reply:
Ok we might as well